docs/tutorial/go.rst

```diff
@@ -0,0 +1,351 @@
+.. _build-a-rock-for-a-go-application:
+
+Build a rock for a Go application
+------------------------------------
+
+In this tutorial, we'll create a simple Go application and learn how to
+containerise it in a rock, using Rockcraft's ``go-framework``
+:ref:`extension <go-framework-reference>`.
+
+Setup
+=====
+
+.. include:: /reuse/tutorial/setup.rst
+
+.. note::
+    This tutorial requires the ``latest/edge`` channel of Rockcraft. Use
+    ``sudo snap refresh rockcraft --channel latest/edge`` to get the latest
+    edge version.
+
+Finally, create a new directory for this tutorial and go inside it:
+
+.. code-block:: bash
+
+   mkdir go-hello-world
+   cd go-hello-world
+
+Create the Go application
+==============================
+
+Start by creating the "Hello, world" Go application that will be used for
+this tutorial.
+
+In order to test the Go application locally (before packing it into a rock),
+install ``go``.
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:install-go]
+    :end-before: [docs:install-go-end]
+    :dedent: 2
+
+Create a ``main.go`` file, copy the following text into it and then
+save it:
+
+.. literalinclude:: code/go/main.go
+    :language: go
+
+Initialise the new go module:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:mod-init-go]
+    :end-before: [docs:mod-init-go-end]
+    :dedent: 2
+
+
+Build the Go application so it can be run:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:go-build]
+    :end-before: [docs:go-build-end]
+    :dedent: 2
+
+A new binary named ``go-hello-world`` should be in the current directory.
+Run the Go application using ``./go-hello-world`` to verify that
+it works.
+
+Test the Go application by using ``curl`` to send a request to the root
+endpoint. You may need a new terminal for this, if you are using Multipass use
+``multipass shell rock-dev`` to get another terminal:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:curl-go]
+    :end-before: [docs:curl-go-end]
+    :dedent: 2
+
+The Go application should respond with ``Hello, world!``.
+
+The Go application looks good, so stop it for now using :kbd:`Ctrl` + :kbd:`C`.
+
+Pack the Go application into a rock
+===================================
+
+
+First, we'll need a ``rockcraft.yaml`` file. Rockcraft will automate its
+creation and tailoring for a Go application by using the
+``go`` profile:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:create-rockcraft-yaml]
+    :end-before: [docs:create-rockcraft-yaml-end]
+    :dedent: 2
+
+The ``rockcraft.yaml`` file will automatically be created in your working
+directory. Open it in a text editor and check that the ``name`` is
```

Comment:
I'm hesitant in showing the file because then we have to keep it up to date with the template. We are explicit about what we say the user should check so I think it should be ok